src/Workspaces/Remote/Core/AbstractAssetProvider.cs

```diff
@@ -104,16 +104,45 @@ async Task<ImmutableArray<DocumentInfo>> CreateDocumentInfosAsync(ChecksumsAndId
         // TODO: do we need version?
         return new DocumentInfo(attributes, textLoader, documentServiceProvider: null);
     }
+}
+
+internal static class AbstractAssetProviderExtensions
+{
+    public static Task GetAssetsAsync<TAsset>(
```

Comment:
lots of hte helpers became extensions.  they don't need to be in the type itself.